test/benchmark/BenchmarkBigIT.java

```diff
@@ -0,0 +1,395 @@
+    private static CoreDatabaseManager databaseMgr;
+
+    @Before
+    public void setUp() throws IOException {
+        Util.resetDirectory(dataDir);
+        databaseMgr = CoreDatabaseManager.open(options);
+        database = "a" + UUID.randomUUID().toString().replaceAll("-", "");
+        databaseMgr.create(database);
+    }
+
+    @After
+    public void tearDown() {
+        databaseMgr.close();
+    }
+
+    private TypeDB.Session schemaSession() {
+        return databaseMgr.session(database, Arguments.Session.Type.SCHEMA);
+    }
+
+    private TypeDB.Session dataSession() {
+        return databaseMgr.session(database, Arguments.Session.Type.DATA);
+    }
+
+    final private Random rand = new Random();
+
+    private void loadOntology(String fileName, TypeDB.Session session) {
+        String filePath = "test/benchmark/resources/" + fileName;
+        try (TypeDB.Transaction tx = session.transaction(Arguments.Transaction.Type.WRITE)) {
+            tx.query().define(com.vaticle.typedb.core.test.benchmark.Util.parseTQL(filePath).asDefine());
+            tx.commit();
+        }
+    }
+
+    private void loadEntities(String entityLabel, int N, TypeDB.Session session) {
+        try (TypeDB.Transaction tx = session.transaction(Arguments.Transaction.Type.WRITE)) {
+            EntityType entityType = tx.concepts().getEntityType(entityLabel);
+            for (int i = 0; i < N; i++) {
+                entityType.create();
+            }
+            tx.commit();
+        }
+    }
+
+    private void loadRandomisedRelationInstances(String entityLabel, String fromRoleLabel, String toRoleLabel,
+                                                 String relationLabel, int N, TypeDB.Session session) {
+        try (TypeDB.Transaction transaction = session.transaction(Arguments.Transaction.Type.WRITE)) {
+            UnboundVariable entity = var("e");
+            Thing[] instances = transaction.query().match(TypeQL.match(entity.isa(entityLabel)))
+                    .map(ans -> ans.get(entity.name()))
+                    .toList().toArray(new Thing[0]);
+
+            assertEquals(instances.length, N);
+            RelationType relationType = transaction.concepts().getRelationType(relationLabel);
+            RoleType fromRole = relationType.getRelates(fromRoleLabel);
+            RoleType toRole = relationType.getRelates(toRoleLabel);
+
+
+            Multimap<Integer, Integer> assignmentMap = HashMultimap.create();
+            for (int i = 0; i < N; i++) {
+                int from = rand.nextInt(N - 1);
+                int to = rand.nextInt(N - 1);
+                while (to == from && assignmentMap.get(from).contains(to)) to = rand.nextInt(N - 1);
+
+                Relation rel = relationType.create();
+                rel.addPlayer(fromRole, instances[from]);
+                rel.addPlayer(toRole, instances[to]);
+            }
+            transaction.commit();
+        }
+    }
+
+    private void loadJoinData(int N) {
```

Comment:
`join` -> `relation`?

Reply:
No, It's the name of the case. 
The join data and transitivity data both contain relations.